### PR TITLE
chore: Address SonarCloud issues in xml-schema-ts

### PR DESCRIPTION
**Pull request by igarashitm**
https://github.com/KaotoIO/kaoto/issues/3008

 - Fixed two bugs in XmlSchemaDerivationMethod.schemaValueOf: the split regex used a literal \s (never matched spaces), and for...in iterated array indices instead of values
 - Reduced cognitive complexity in SchemaBuilder, XmlSchema, and XmlSchemaDerivationMethod by extracting focused helper methods (applyToken, findByQName, applyComplexTypeAttributes, handleElementConstraints, collectExtensionAttributes) and adding early-return guards
 - Replaced the 6 nearly-identical getXxxByQName methods in XmlSchema with a single generic findByQName<T> helper
 - Added readonly to fields that are initialized once and never reassigned across SchemaBuilder, XmlSchema, XmlSchemaCollection, and SchemaKey
 - Replaced == null / != null with strict === null / !== null (or ?? / ?? =) throughout the module and subdirectories
 - Added unit tests for XmlSchemaDerivationMethod edge cases and integration tests covering final, block, and abstract derivation method parsing through real XSD fixtures

## Summary by CodeRabbit

* **New Features**
  * Added built-in DerivationMethods and Constraints XML schemas for richer schema scenarios.

* **Refactor**
  * Hardened internal schema handling with stricter null/undefined checks and increased immutability to improve parsing reliability and namespace/name resolution.

* **Tests**
  * Expanded test coverage for parsing, derivation semantics, constraints, extension registry dispatch, and namespace utilities.

### CODE REVIEW

File: packages/ui/src/xml-schema-ts/SchemaBuilder.ts [hunk 1]
```diff
@@ -863,39 +786,75 @@ export class SchemaBuilder {
...
+  }
+
+  private handleElementConstraints(element: XmlSchemaElement, el: Element): void {
+    let keyEl = XDOMUtil.getFirstChildElementNS(el, XmlSchema.SCHEMA_NS, 'key');
+    while (keyEl !== null) {
+      element.getConstraints().push(this.handleConstraint(keyEl, new XmlSchemaKey()));
+      keyEl = XDOMUtil.getNextSiblingElementNS(keyEl, XmlSchema.SCHEMA_NS, 'key');
+    }
+    let keyrefEl = XDOMUtil.getFirstChildElementNS(el, XmlSchema.SCHEMA_NS, 'keyref');
+    while (keyrefEl !== null) {
```
Comment (mmelko) [inline packages/ui/src/xml-schema-ts/SchemaBuilder.ts:821]: in the original code there was: 
``` 
if (keyrefEl.hasAttribute('refer')) {
          const name = keyrefEl.getAttribute('refer')!;
          keyRef.refer = this.getRefQName(name, el);
}
``` 
isn't this still necessary?  it looks like keyRef.refer  is never set now.

Comment (igarashitm) [reply packages/ui/src/xml-schema-ts/SchemaBuilder.ts:821]: refer attribute is handled in following handleConstraint() separately

Comment (mmelko) [reply packages/ui/src/xml-schema-ts/SchemaBuilder.ts:821]: ah that's true. Thanks!
